public/js/charts/trendsCoverage.js

```diff
@@ -0,0 +1,173 @@
+// NOTE: requires('sheets');
+// NOTE: requires('aq');
+
+window.aq.charts.trendsCoverage = {};
+
+window.Chart.defaults.global.defaultFontSize = 10;
+
+const acquisition = document.getElementById('acquisition');
+
+async function handleLocationChange() {
+  const nodes = await window.aq.getNodes();
+  const countyCitiesMap = await window.sheets.getCountyCitiesMap();
+  function getAvg(sensor) {
+    const total = sensor.reduce((acc, c) => acc + c, 0);
+    return total / sensor.length;
+  }
+  // Format dates from nodes
+  function formatDate(date, langCode) {
+    const dateObj = new Date(date);
+    const day = dateObj.toLocaleString(langCode, { day: '2-digit' }); // DD
+    const month = dateObj.toLocaleString(langCode, { month: 'short' }); // MMM
+    return `${day} ${month}`;
+  }
+  // get chartlabels for the week in an array
+  const labels = [...Array(7)]
+    .map((_, i) => {
+      const d = new Date();
+      d.setDate(d.getDate() - i);
+      return d;
+    })
+    .map((item) => formatDate(item, 'en-US'))
+    .slice()
+    .sort((a, b) => new Date(a).getTime() - new Date(b).getTime());
+
+  const allCountyNodes = Object.entries(countyCitiesMap)
+    .map(([countyName, countyCities]) => {
+      const countyNodes = nodes.filter(({ location }) =>
+        countyCities.some(
+          (city) =>
+            location && location.city && location.city.indexOf(city) >= 0
+        )
+      );
+      return { name: countyName, value: countyNodes };
+    })
+    // Don't show counties with 0 nodes
+    .filter(({ value }) => value.length !== 0)
+    .flatMap((node) => node?.value)
+    .flatMap((node) => node?.sensors)
+    .flatMap((node) => node?.sensordatas)
+    // eslint-disable-next-line func-names
+    .reduce(function (h, obj) {
+      // eslint-disable-next-line no-param-reassign
+      h[obj?.timestamp.split('T')[0]] = (
+        h[obj?.timestamp.split('T')[0]] || []
+      ).concat(obj);
+      return h;
+    }, {});
+
+  const getDateAndSensorTypes = Object.entries(allCountyNodes).map((node) => {
+    return {
+      date: formatDate(node[0], 'en-US'),
+      sensorTypes: node[1]
+        .flatMap((sensor) => sensor?.sensordatavalues)
+        // eslint-disable-next-line func-names
+        .filter((item) => item?.value_type === 'P2')
+        // eslint-disable-next-line func-names
+        .reduce(function (h, obj) {
+          // eslint-disable-next-line no-param-reassign
+          h[obj?.value_type] = (h[obj?.value_type] || []).concat(obj);
+          // eslint-disable-next-line no-console
+          return h;
+        }, {}),
+    };
+  });
+  const filteredP2Data = getDateAndSensorTypes.filter((item) =>
+    labels.find((data) =>
+      data?.toLowerCase().trim().includes(item.date?.toLowerCase().trim())
+    )
+  );
```

Comment:
Not sure I follow what's happening here.

Reply:
@kilemensi So for this, I am grouping data by date in `getDateAndSensorTypes` ...um, this is going to returning something like 👇🏽 . However, the data we have is  not returning the current week data, its returning data for 7 days, for example, we have data values that are one month apart as such 👇🏽 
<img width="1136" alt="Screenshot 2021-06-29 at 11 16 01" src="https://user-images.githubusercontent.com/3693563/123762844-dc3edc80-d8cb-11eb-9e88-21efc8082621.png">

To get data that reflect the current data for the week, I got labels values here  and used that to filter out data not that is not for the current week using `filteredP2Data`

https://github.com/CodeForAfrica/sensors.AFRICA-Dashboard-KE-HAP/blob/face8b2562298de1b4f8aec7946d9a55969f39e4/public/js/charts/trendsCoverage.js#L23